src/ucp/core/ucp_worker.c

```diff
@@ -295,6 +296,13 @@ ucp_worker_iface_error_handler(void *arg, uct_ep_h uct_ep, ucs_status_t status)
         ucp_ep->uct_eps[failed_lane] = NULL;
     }
 
+    /* NOTE: if failed ep is stub auxiliary then we need to replace the lane
+     *       with failed ep and destroy stub ep
+     */
+    if (ucp_stub_ep_test(ucp_ep_iter->uct_eps[0])) {
+        ucp_stub_ep_aux_failed(&ucp_ep_iter->uct_eps[0]);
```

Comment:
i'd suggest to update the code in line 262 for better encapsulation, so you would pass uct_ep to stub_ep: 
`ucp_stub_ep_test_aux(ucp_ep_iter->uct_eps[lane], uct_ep)`
this function will return 1 if the 1st argument is a stub ep, and uct_ep equals to its internal aux_ep

and also rename `ucp_stub_ep_aux_failed` to uct_ep_h ucp_stub_ep_extract_aux(stub_ep)`